### PR TITLE
feat(app): Add error modals to OT-2 calibration flows

### PR DESCRIPTION
**Pull request by mjhuff**
Closes [RQA-3273](https://opentrons.atlassian.net/browse/RQA-3273)

# Overview

RESC tickets identify that a common problem for OT-2 users undergoing calibration flows is that there is not any error messaging when something goes wrong. More specifically, the app never bubbles up errors received from the server, effectively black-holing them.

This PR adds an error details modal modeled after [one used in drop tip wizard](https://www.figma.com/design/Nx7ORMnfyJNP4FQYxN4e5x/Primary%3A-Desktop-App?node-id=5283-145024&t=tC8MqikTjztZrbIk-4), implementing it for all OT-2 calibration flows. Because stale errors will display after relaunching calibration flows after an error, we clear all `robotApi` requests every time we launch a calibration flow. Doing so should not impact the app, since the selectors utilizing these requests were unused until now! Note: clearing just the one error request is insufficient, since it's possible for a user to receive several of the same errored responses, which results in the same stale error problem on the next calibration flow. 

https://github.com/user-attachments/assets/13730210-58e4-43be-a86f-b33ebd9974c6

## Test Plan and Hands on Testing
- See video.

## Changelog
- Added error modals to OT-2 calibration flows.

## Review requests
@ecormany I've specified some fallback copy defined in `app/src/assets/localization/en/robot_calibration.json`. Is this sufficient or would you like me to change anything?

## Risk assessment
lowish-medium. We can ask QA to test calibration flows to be safe!

[RQA-3273]: https://opentrons.atlassian.net/browse/RQA-3273?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/src/assets/localization/en/robot_calibration.json [hunk 1]
```diff
@@ -122,6 +124,7 @@
   "tip_pick_up_instructions": "<block>Using the controls below or your keyboard, jog the pipette until the nozzle closest to you is centered above the A1 position and level with the top of the tip.</block><br/><block>When the pipette is properly aligned, pick up the tip.</block>",
   "title": "robot calibration",
   "to_check": "To check the {{mount}} pipette:",
+  "unexpected_error": "An unexpected error has occurred. If the issue persists, contact customer support for assistance.",
```
Comment (ecormany) [inline app/src/assets/localization/en/robot_calibration.json]: We would have this be `Opentrons Support` instead of `customer support` in branded copy.

**Review by ecormany — COMMENTED** [review]
Only reviewed localization file, left one comment.

Also noticed in the video that we have a floating point error in the error text. Not sure whether we want to bother spending effort to truncate that decimal.

File: app/src/assets/localization/en/robot_calibration.json (hunk 1 above)
Comment (mjhuff) [reply app/src/assets/localization/en/robot_calibration.json]: Ahhhh you're totally right. Good catch, thanks!

**Review by sfoster1 — APPROVED** [review]
Bless

**Pull request merged**